File: app/src/main/java/com/amir/alzheimer/base/BaseFragment.java
package com.amir.alzheimer.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.amir.alzheimer.infrastructure.Database;

public abstract class BaseFragment extends Fragment {
    protected AlzheimerApplication application;
    protected Database database;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        application = (AlzheimerApplication) getActivity().getApplication();
        database = application.getDatabase();
    }
}

File: app/src/main/java/com/amir/alzheimer/base/BaseFragment.kt
package com.amir.alzheimer.base


import android.os.Bundle
import androidx.fragment.app.Fragment

abstract class BaseFragment : Fragment() {
    protected lateinit var application: AlzheimerApplication

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.application = activity!!.application as AlzheimerApplication
    }
}
